feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/mixin/maxAction/SwapExistentialDepositAwareMaxActionProvider.kt

```diff
@@ -0,0 +1,63 @@
+package io.novafoundation.nova.feature_swap_impl.presentation.mixin.maxAction
+
+import io.novafoundation.nova.common.utils.orFalse
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.AssetSourceRegistry
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.existentialDepositInPlanks
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.isSelfSufficientAsset
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.types.Balance
+import io.novafoundation.nova.feature_wallet_api.domain.model.Asset
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.amountChooser.maxAction.MaxActionProvider
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.amountChooser.maxAction.minus
+import io.novafoundation.nova.runtime.ext.fullId
+import io.novafoundation.nova.runtime.ext.isCommissionAsset
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+
+class SwapExistentialDepositAwareMaxActionProvider(
+    private val assetOutFlow: Flow<Asset?>,
+    private val assetSourceRegistry: AssetSourceRegistry,
+    private val inner: MaxActionProvider,
+    chainRegistry: ChainRegistry,
+) : MaxActionProvider {
+
+    // Fee is not deducted for display
+    override val maxAvailableForDisplay: Flow<Balance?> = inner.maxAvailableForDisplay
+
+    override val maxAvailableForAction: Flow<MaxActionProvider.MaxAvailableForAction?> = combine(
+        inner.maxAvailableForAction.filterNotNull(),
+        getAssetInTotalCanDropBelowMinimumBalanceFlow(),
+        assetOutSelfSufficiency()
+    ) { maxAvailable, canDropBelowMinimumBalance, assetOutSelfSufficiency ->
+        if (maxAvailable.chainAsset.isCommissionAsset && (canDropBelowMinimumBalance || !assetOutSelfSufficiency)) {
```

Comment:
Shoudlnt it be `maxAvailable.chainAsset.isCommissionAsset && (!canDropBelowMinimumBalance || !assetOutSelfSufficiency)` (added negation for canDropBelowMinimumBalance)?
In any case, lets split this condition into multiple smaller parts for easier understanding

Reply:
Yes, thank you!